src/inertia.js

```diff
@@ -119,20 +131,25 @@ export default {
       if (visitId === this.visitId) {
         this.version = page.version
         this.setState(page, replace, preserveState)
-        this.updatePage(component, page.props, { preserveState })
-        this.setScroll(preserveScroll)
+        this.updatePage(component, page.props, { preserveState }).then(() => {
+          let scrollRegions = document.querySelectorAll('[scroll-region]')
+
+          scrollRegions.forEach(region => {
+            region.addEventListener('scroll', this.saveScrollPositions)
+          })
+
+          if (!preserveScroll) {
+            window.scrollTo(0, 0)
+            scrollRegions.forEach(region => region.scrollTo(0, 0))
```

Comment:
@reinink I think it would probably be better to use `scrollTop` rather than `scrollTo`? This will fix the cross compatibility issues with MS Edge. For example:

```javascript
window.scrollTop = 0
scrollRegions.forEach(region => region.scrollTop = 0)
```

Reply:
Yup, done. 👍 

//cc @marvinrabe